lib/play.js

```diff
@@ -44,11 +49,32 @@ const makeTurn = async (store) => {
   }
 };
 
+const recieveGameSetupChoices = async (store) => {
+  return new Promise((resolve) => {
+    store.ws.on("message", (message) => {
```

Comment:
(nice-to-have refactor that can provably wait): I feel like this pattern of: 
```
  return new Promise((resolve) => {
    store.ws.on("message", (message) => {
...
```
is a bit of a mind-fuck, and is repeated in quite a few places. Could you make a single `receiveMsg(ws)` helper function that does this once, and then `let msg = receiveMsg(ws)` in a bunch of places instead?